### PR TITLE
Feature: iPad - support drag and drop an image from Safari.

### PR DESCRIPTION
**Pull request by KaterinaWire**
## What's new in this PR?

Add `UIDropInteractionDelegate` to the `ConversationInputBarViewController` and implement 3 methods for dragging and dropping images.

### CODE REVIEW

File: Wire-iOS.xcodeproj/project.pbxproj [hunk 1]
```diff
@@ -8,6 +8,7 @@
 
 /* Begin PBXBuildFile section */
 		061282632379C25500C1A53C /* UITextView+ReplaceTests.swift in Sources */ = {isa = PBXBuildFile; fileRef = 061282622379C25500C1A53C /* UITextView+ReplaceTests.swift */; };
+		06C412B0239103910018866F /* ConversationInputBarViewController+Drag&Drop.swift in Sources */ = {isa = PBXBuildFile; fileRef = 06C412AF239103910018866F /* ConversationInputBarViewController+Drag&Drop.swift */; };
```
Comment (billypchan) [inline Wire-iOS.xcodeproj/project.pbxproj]: I would suggest not to use & in file name

File: Wire-iOS/Sources/UserInterface/Conversation/InputBar/ConversationInputBarViewController.m [hunk 2]
```diff
@@ -91,6 +91,10 @@ @interface ConversationInputBarViewController (GiphySearchViewController) <Giphy
 
 @end
 
+@interface ConversationInputBarViewController (UIDropInteractionDelegate) <UIDropInteractionDelegate>
+
+@end
+
```
Comment (billypchan) [inline Wire-iOS/Sources/UserInterface/Conversation/InputBar/ConversationInputBarViewController.m]: This is not needed

File: Wire-iOS/Sources/UserInterface/Conversation/InputBar/ConversationInputBarViewController+Drag&Drop.swift [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+    }
+    
+    @available(iOS 11.0, *)
+    public func dropInteraction(_ interaction: UIDropInteraction, sessionDidUpdate session: UIDropSession) -> UIDropProposal {
+        return UIDropProposal(operation: .copy)
+    }
+    
+    @available(iOS 11.0, *)
+    public func dropInteraction(_ interaction: UIDropInteraction, canHandle session: UIDropSession) -> Bool {
+        return session.canLoadObjects(ofClass: UIImage.self)
```
Comment (billypchan) [inline Wire-iOS/Sources/UserInterface/Conversation/InputBar/ConversationInputBarViewController+Drag&Drop.swift]: Can we drop ant drop file or text? If no we can do it in another PR.

File: Wire-iOS/Sources/UserInterface/Conversation/InputBar/ConversationInputBarViewController+Drag&Drop.swift [hunk 4]
```diff
@@ -0,0 +1,66 @@
...
+
+extension ConversationInputBarViewController: UIDropInteractionDelegate {
+    
+    @available(iOS 11.0, *)
+    public func dropInteraction(_ interaction: UIDropInteraction, performDrop session: UIDropSession) {
+        
+        for dragItem in session.items {
+            dragItem.itemProvider.loadObject(ofClass: UIImage.self, completionHandler: { object, error in
+                
+                guard error == nil else { return print("Failed to load our dragged item") }
```
Comment (billypchan) [inline Wire-iOS/Sources/UserInterface/Conversation/InputBar/ConversationInputBarViewController+Drag&Drop.swift]: we can log the error with `zmLog` and the error should be printed.

File: Wire-iOS/Sources/UserInterface/Conversation/InputBar/ConversationInputBarViewController+Drag&Drop.swift [hunk 5]
```diff
@@ -0,0 +1,66 @@
...
+                            if let draggedImageData = draggedImage.pngData() {
+                                self.sendController.sendMessage(withImageData: draggedImageData, completion: .none)
+                            }
+                        }
+                    }
+                    
+                    confirmImageViewController.onCancel = { [unowned self] in
+                        self.dismiss(animated: true)
+                    }
+                    self.present(confirmImageViewController, animated: true) {
```
Comment (billypchan) [inline Wire-iOS/Sources/UserInterface/Conversation/InputBar/ConversationInputBarViewController+Drag&Drop.swift]: What happens when multiple images are dropped? The last one would appear first?

Comment (KaterinaWire) [reply Wire-iOS/Sources/UserInterface/Conversation/InputBar/ConversationInputBarViewController+Drag&Drop.swift]: I can't reproduce it on a simulator (to select several images), but we check each element in the array of selected images (`session.items`). So, a new `ConfirmAssetViewController` will be opened for each image and the last image in the array will be displayed first.

File: Wire-iOS/Sources/UserInterface/Conversation/InputBar/ConversationInputBarViewController+DragAndDrop.swift [hunk 6]
```diff
@@ -0,0 +1,68 @@
...
+
+extension ConversationInputBarViewController: UIDropInteractionDelegate {
+    
+    @available(iOS 11.0, *)
+    public func dropInteraction(_ interaction: UIDropInteraction, performDrop session: UIDropSession) {
+        
+        for dragItem in session.items {
+            dragItem.itemProvider.loadObject(ofClass: UIImage.self, completionHandler: { object, error in
+                
+                guard error == nil else { return zmLog.error("Failed to load our dragged item") }
```
Comment (billypchan) [inline Wire-iOS/Sources/UserInterface/Conversation/InputBar/ConversationInputBarViewController+DragAndDrop.swift]: We can include `error` in the log message.
What kind of errors we may encounter? Do we need to show a alert of some kind of errors to help the user to solve the issues?

File: Wire-iOS/Sources/UserInterface/Conversation/InputBar/ConversationInputBarViewController+Drag&Drop.swift (hunk 5 above)
Comment (KaterinaWire) [reply Wire-iOS/Sources/UserInterface/Conversation/InputBar/ConversationInputBarViewController+Drag&Drop.swift]: When multiple images are dropped, only one ConfirmAssetViewController is displayed with the first image.

Comment (KaterinaWire) [reply Wire-iOS/Sources/UserInterface/Conversation/InputBar/ConversationInputBarViewController+Drag&Drop.swift]: Is it fine?

Comment (billypchan) [reply Wire-iOS/Sources/UserInterface/Conversation/InputBar/ConversationInputBarViewController+Drag&Drop.swift]: Then let's discuss with designer with this tickets

https://github.com/wireapp/wire-ios/pull/3460

File: Wire-iOS/Sources/UserInterface/Conversation/InputBar/ConversationInputBarViewController+DragAndDrop.swift [hunk 7]
```diff
@@ -0,0 +1,68 @@
...
+                            if let draggedImageData = draggedImage.pngData() {
+                                self.sendController.sendMessage(withImageData: draggedImageData, completion: .none)
+                            }
+                        }
+                    }
+                    
+                    confirmImageViewController.onCancel = { [unowned self] in
+                        self.dismiss(animated: true)
+                    }
+                    self.present(confirmImageViewController, animated: true) {
```
Comment (billypchan) [inline Wire-iOS/Sources/UserInterface/Conversation/InputBar/ConversationInputBarViewController+DragAndDrop.swift]: If we want to support one image for now, we should break after a `ConfirmAssetViewController` is presented.

File: Wire-iOS/Sources/UserInterface/Conversation/InputBar/ConversationInputBarViewController+Drag&Drop.swift (hunk 5 above)
Comment (KaterinaWire) [reply Wire-iOS/Sources/UserInterface/Conversation/InputBar/ConversationInputBarViewController+Drag&Drop.swift]: Ok
So, as part of this ticket, we will drop only one image.

File: Wire-iOS/Sources/UserInterface/Conversation/InputBar/ConversationInputBarViewController+DragAndDrop.swift (hunk 6 above)
Comment (KaterinaWire) [reply Wire-iOS/Sources/UserInterface/Conversation/InputBar/ConversationInputBarViewController+DragAndDrop.swift]: I added `error` to the log message. 
However, at the moment, I have no idea what issues the `loadObject` method can cause that the user can solve.

File: Wire-iOS/Sources/UserInterface/Conversation/InputBar/ConversationInputBarViewController+DragAndDrop.swift (hunk 7 above)
Comment (KaterinaWire) [reply Wire-iOS/Sources/UserInterface/Conversation/InputBar/ConversationInputBarViewController+DragAndDrop.swift]: I added `break` in the loop.

File: Wire-iOS/Sources/UserInterface/Conversation/InputBar/ConversationInputBarViewController+Drag&Drop.swift (hunk 3 above)
Comment (KaterinaWire) [reply Wire-iOS/Sources/UserInterface/Conversation/InputBar/ConversationInputBarViewController+Drag&Drop.swift]: We can drop text and link, but I can't create a file on a simulator.
